packages/runtime/container-runtime/src/orderedClientElection.ts

```diff
@@ -335,10 +402,16 @@ export class OrderedClientElection
     private addClient(client: ILinkedClient, sequenceNumber: number): void {
         if (this.isEligibleFn(client)) {
             this._eligibleCount++;
-            if (this._electedClient === undefined) {
-                // Automatically elect latest client
+            const isSummarizer = client.client.details.type === summarizerClientType;
+            const electedClientIsSummarizer = this._electedClient?.client.details.type === summarizerClientType;
+            // Note that we allow a summarizer client to supercede an interactive client as elected client.
+            if (this._electedClient === undefined || (isSummarizer && !electedClientIsSummarizer)) {
```

Comment:
nit: Making the second part of this if statement (!electedClientIsSummarizer && isSummarizer) will read better. Basically, the elected client is not summarizer but the added client is.